src/commands/stepfunctions/helpers.ts

```diff
@@ -159,18 +159,8 @@ export type ParametersType = {
 //      | "Payload.$": "$$['Execution', 'State', 'StateMachine']"  |
 //  4.3 | Custom "Payload.$"                                       | false
 export const injectContextForLambdaFunctions = (step: StepType, context: BaseContext, stepName: string): boolean => {
-  if (step.Resource?.startsWith('arn:aws:lambda')) {
-    context.stdout.write(
-      `[Warn] Step ${stepName} may be using the basic legacy integration, which does not support merging lambda trace(s) with Step Functions trace.
-          To merge lambda trace(s) with Step Functions trace, please consider using the latest integration.
-          More details can be found on https://docs.aws.amazon.com/step-functions/latest/dg/connect-lambda.html \n`
-    )
-
-    return false
-  }
-
-  // not default lambda api
-  if (step.Resource !== 'arn:aws:states:::lambda:invoke') {
+  // not default lambda api or legacy lambda definition
+  if (step.Resource !== 'arn:aws:states:::lambda:invoke' && !step.Resource?.startsWith('arn:aws:lambda')) {
```

Comment:
There's also the [callback invocation pattern](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-wait-token) for invoking the lambda which probably still won't work (If you're thinking of handling that in a separate PR, that's fine too). This is a case where the value for `step.Resource` will be `arn:aws:states:::lambda:invoke.waitForTaskToken`